src/transformers/modeling_tf_utils.py

```diff
@@ -247,12 +247,83 @@ def compute_loss(self, labels, logits):
         return loss_fn(next_sentence_label, next_sentence_reduced_logits)
 
 
-def input_processing(func, input_ids, **kwargs):
+def booleans_processing(config, **kwargs):
+    """
+    Process the input booleans of each model in order to be sure they are compliant with the execution mode (eager or
+    graph)
+
+    Args:
+        config (:class:`~transformers.PretrainedConfig`):
+            The config of the running model.
+        **kwargs:
+            The boolean parameters
+
+    Returns:
+        A dictionary with the proper values for each boolean
+    """
+    final_booleans = {}
+
+    if tf.executing_eagerly():
+        final_booleans["output_attentions"] = (
+            kwargs["output_attentions"] if kwargs["output_attentions"] is not None else config.output_attentions
+        )
+        final_booleans["output_hidden_states"] = (
+            kwargs["output_hidden_states"]
+            if kwargs["output_hidden_states"] is not None
+            else config.output_hidden_states
+        )
+
+        if "return_dict" in kwargs:
+            final_booleans["return_dict"] = (
+                kwargs["return_dict"] if kwargs["return_dict"] is not None else config.return_dict
+            )
+
+        if "use_cache" in kwargs:
+            final_booleans["use_cache"] = kwargs["use_cache"] if kwargs["use_cache"] is not None else config.use_cache
+    else:
+        if (
+            kwargs["output_attentions"] is not None
+            or kwargs["output_hidden_states"] is not None
+            or ("use_cache" in kwargs and kwargs["output_hidden_states"] is not None)
+            or ("return_dict" in kwargs and kwargs["return_dict"] is not None)
+        ):
+            logger.warn(
+                "Cannot update the boolean parameters behavior in graph mode and the return_dict parameter is always True in that mode."
```

Comment:
I'd make the warning a bit explicit and split it into 2:
1) Cannot set boolean arguments `use_cache`, `output_hidden_states`, and `output_attentions` to True in graph mode
and then under new if `("return_dict" in kwargs and kwargs["return_dict"] is not None)`
2) `return_dict` is always set to True in graph mode

Reply:
BTW you can have `use_cache`, `output_hidden_states` and `output_attentions` to `True` in graph mode, but it has to be done when instantiating the `config`.

Reply:
Mayne we can make that more explicit in the warning then? I hadn't understood it from the current one,